### PR TITLE
Countable java test

### PR DESCRIPTION
**Pull request by nastydrew**

### CODE REVIEW

File: Java/Countable/test/CountableTest.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public void CountTest() throws Throwable {
+        Countable obj1 = new Countable();
+        Countable obj2 = new Countable();
+        Countable obj3 = new Countable();
+
+        Assert.assertEquals(Countable.getTotalCount(), 3);
+
+        obj2.destroy();
+        Assert.assertEquals(Countable.getTotalCount(), 2);
+    }
```
Comment (igordotsenko) [inline Java/Countable/test/CountableTest.java:28]: You should think about all edge cases. For example, for this class edge case is when all `Countable` objects are destroyed and after that you will try to destroy it again. What should happen in this case?

File: Java/Countable/src/Countable.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        totalCount += 1;
+    }
+
+
+    public void destroy() throws Throwable{
+        totalCount -= 1;
+
+        if ( totalCount == 0 ) {
+            throw new NoCountableObjectException();
+        }
```
Comment (igordotsenko) [inline Java/Countable/src/Countable.java]: I guess this check should be before decrementing of `totalCount`

File: Java/Countable/test/CountableTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    public void countTest() throws Throwable {
+        Countable obj1 = new Countable();
+        Countable obj2 = new Countable();
+        Countable obj3 = new Countable();
+
+        Assert.assertEquals(Countable.getTotalCount(), 3);
+        obj3.destroy();
+        Assert.assertEquals(Countable.getTotalCount(), 2);
+        obj1.destroy();
+        obj2.destroy();
```
Comment (igordotsenko) [inline Java/Countable/test/CountableTest.java]: Redundant work.

**Review by igordotsenko — CHANGES_REQUESTED** [review]

**Review by igordotsenko — APPROVED** [review]
:shipit:
